### PR TITLE
Data Access Interface Overhaul + minor edits to my own code (renaming attributes)

### PR DESCRIPTION
**Pull request by daniyaalf**
**Copy of my commit**: I created a new package in use_case for data access interfaces. Then, I renamed ethan's DataAccessInterface to GameDataAccessInterface and I removed the getPromptGenerator() method so that it only has save(Game) and getGame(). This class can be used by use cases that don't need to generate stories or prompts etc. I made another class called PromptGameDataAccessInterface that extends GameDataAccessInterface and it contains a method getPromptGenerator() and a method save(PromptGenerator). Since it extends the other class it has the game related methods too. This is the interface my vote out use case and Ethan's kill villager use case should be using.

**EDIT**: Making the prompt/game data access interfaces separate after Ethan noticed we can just implement both (better single responsibility principle usage)

I also renamed some attributes in my own files to be more consistent with Ethan's.

Lastly, I deleted the VoteOutDataAccessInterface class.

### CODE REVIEW

File: src/data_access/GameDataAccessObject.java
Comment (yhz3) [inline src/data_access/GameDataAccessObject.java:1]: From the name I assumed this DAO is supposed to getGame() and saveGame(), so if it's supposed to save the prompts/conversation history, should we change the name to PromptDataAccessObject?

File: src/use_case/data_access_interface/ChatAPIAccessInterface.java
Comment (yhz3) [inline src/use_case/data_access_interface/ChatAPIAccessInterface.java:1]: I think it's in the right place, just maybe add in the getResponse()

File: src/use_case/vote_out/VoteOutInteractor.java [hunk 1]
```diff
@@ -2,19 +2,20 @@
...
 
 
-    public VoteOutInteractor(VoteOutDataAccessInterface gameDataAccessObject, ChatAPIAccessInterface storyDataAccessObject, VoteOutOutputBoundary userPresenter) {
+    public VoteOutInteractor(PromptGameDataAccessInterface gameDataAccessObject, ChatAPIAccessInterface gptDataAccessObject, VoteOutOutputBoundary userPresenter) {
         this.gameDataAccessObject = gameDataAccessObject;
-        this.storyDataAccessObject = storyDataAccessObject;
+        this.gptDataAccessObject = gptDataAccessObject;
         this.userPresenter = userPresenter;
         // Get the game
         game = gameDataAccessObject.getGame();
```
Comment (yhz3) [inline src/use_case/vote_out/VoteOutInteractor.java:21]: From what I can see PromptGameDataAccess does not have a getGame() method, correct me if I'm wrong but it seems to be a refactoring mistake?

File: src/data_access/GameDataAccessObject.java
Comment (daniyaalf) [reply src/data_access/GameDataAccessObject.java:1]: yeah i thought about it though in java the naming convention when you extend classes is usually to add a word before the entire name of the class you're extending

File: src/use_case/vote_out/VoteOutInteractor.java (hunk 1 above)
Comment (daniyaalf) [reply src/use_case/vote_out/VoteOutInteractor.java:21]: it automatically has the methods in the class it's extending, check my interactor it works

File: src/use_case/data_access_interface/ChatAPIAccessInterface.java
Comment (daniyaalf) [reply src/use_case/data_access_interface/ChatAPIAccessInterface.java:1]: It has it lol github is just not showing the entire class there should be an arrow that lets you see the rest

File: src/data_access/GameDataAccessObject.java
Comment (daniyaalf) [reply src/data_access/GameDataAccessObject.java:1]: (Disregard now that I'm splitting the interfaces up)

Comment (yhz3) [reply src/data_access/GameDataAccessObject.java:1]: Commenting this here so it is on record that it was discussed we would not have PromptDAO extend GameDAO and just have PromptDAO be used for saving and getting prompts while GameDAO is only used for saving and getting the game, adhering to SRP better.

Comment (daniyaalf) [reply src/data_access/GameDataAccessObject.java:1]: Yeah added it to pull request description too

File: src/use_case/data_access_interface/ConversationDataAccessInterface.java [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 import entity.PromptGenerator;
 
-public interface PromptDataAccessInterface {
+public interface ConversationDataAccessInterface {
     PromptGenerator getPromptGenerator();
     // getPromptGenerator in DAO should be pulling the ConversationHistory to then construct a PromptGenerator off of
     void save(PromptGenerator promptGenerator);
```
Comment (yhz3) [inline src/use_case/data_access_interface/ConversationDataAccessInterface.java:8]: Change this to Conversation History entity. Within other classes, we can just do promptGenerator.getConversationHistory() to get ConversationHistory.

File: src/use_case/kill_villager/KillVillagerDataAccessInterface.java
Comment (yhz3) [inline src/use_case/kill_villager/KillVillagerDataAccessInterface.java:1]: This file should just be deleted from the codebase

**Review by yhz3 — APPROVED** [review]
Kill Villager use case not reviewed

**Pull request merged**